src/components/Graph.react.js

```diff
@@ -21,20 +21,44 @@ const filterEventData = (gd, eventData, event) => {
             const pointData = filter(function(o) {
                 return !contains(type(o), ['Object', 'Array'])
             }, fullPoint);
-
-            if (has('customdata', data[pointData.curveNumber]) &&
+            if (has('curveNumber', fullPoint) &&
                 has('pointNumber', fullPoint) &&
-                has('curveNumber', fullPoint)
-            ) {
-                pointData['customdata'] = data[
-                    pointData.curveNumber
-                ].customdata[fullPoint.pointNumber];
+                has('customdata', data[pointData.curveNumber])) {
+               pointData['customdata'] = data[
+                                            pointData.curveNumber
+                                        ].customdata[fullPoint.pointNumber];
             }
 
-            points[i] = pointData;
+            /*
+              Works to assign customdata values to sankey and pie graph
+              this only works if there is one trace per plot otherwise
+              it begins to repeat customdata for different graph or potentially
+              shows incorrect customdata
+            */
+            if(gd.data.length < 2){
+              const traceNumber = 0;
+              const traceType =  gd.calcdata[0][0]['trace'].type;
+              if((traceType === 'sankey' | traceType == 'pie') &&
+                  has('customdata', gd.data[traceNumber])){
+                var pointNumber = 0;
+                if(traceType === 'sankey') {
+                  pointNumber = gd.data[traceNumber]
+                                          ['node']
+                                          ['label'].indexOf(pointData['label']);
+                } else {
+                  pointNumber = pointData.i;
+                }
+
+                pointData['customdata'] = gd.data[
+                                                traceNumber
+                                              ].customdata[pointNumber];
+                pointData['index'] = pointNumber;
+              }
+            }
+              points[i] = pointData;
+          }
```

Comment:
it seems to me that this is just a bug in plotly.js: `pointNumber` and `curveNumber` should be included in the `pointData` for all the chart types, including `pie` and `sankey`. Let's remove this block for now and can you open an issue in `plotly.js` with a reproducable example? https://github.com/plotly/plotly.js/issues/new